src/tests/multihost/adsites/conftest.py

```diff
@@ -0,0 +1,305 @@
+        session_multihost.client[0].run_command(restore)
+    request.addfinalizer(restoresssdconf)
+
+
+@pytest.fixture(scope='function')
+def create_site(session_multihost, request):
+    ad2_hostname = session_multihost.ad[1].hostname
+    ad2_shostname = ad2_hostname.strip().split('.')[0]
+    site = "Raleigh"
+
+    cmd_create_site = "powershell.exe -inputformat none -noprofile " \
+                      "'(New-ADReplicationSite -Name \"%s\" " \
+                      "-Confirm:$false)'" % site
+    cmd_move_ad2 = "powershell.exe -inputformat none -noprofile " \
+                   "'(Move-ADDirectoryServer -Identity \"%s\" -Site \"%s\" " \
+                   "-Confirm:$false)'" % (ad2_shostname, site)
+
+    session_multihost.ad[0].run_command(cmd_create_site)
+    session_multihost.ad[0].run_command(cmd_move_ad2)
+
+    def teardown_site():
+        cmd_move_ad2back = "powershell.exe -inputformat none -noprofile " \
+                           "'(Move-ADDirectoryServer -Identity \"%s\" " \
+                           "-Site \"Default-First-Site-Name\" " \
+                           "-Confirm:$false)'" % ad2_shostname
+        cmd_remove_site2 = "powershell.exe -inputformat none -noprofile " \
+                           "'(Remove-ADReplicationSite \"%s\" " \
+                           "-Confirm:$false)'" % site
+        session_multihost.ad[0].run_command(cmd_move_ad2back)
+        session_multihost.ad[0].run_command(cmd_remove_site2)
+
+    request.addfinalizer(teardown_site)
+
+
+# ############## class scoped Fixtures ##############################
+
+
+@pytest.fixture(scope="class")
+def multihost(session_multihost, request):
+    """ Multihost fixture to be used by tests
+    :param obj session_multihost: multihost object
+    :return obj session_multihost: return multihost object
+    :Exceptions: None
+    """
+    if hasattr(request.cls(), 'class_setup'):
+        request.cls().class_setup(session_multihost)
+        request.addfinalizer(
+            lambda: request.cls().class_teardown(session_multihost))
+    return session_multihost
+
+
+@pytest.fixture(scope="class")
+def clear_sssd_cache(session_multihost):
+    """ Clear sssd cache """
+    client = sssdTools(session_multihost.client[0])
+    client.clear_sssd_cache()
+
+
+@pytest.fixture(scope="class")
+def joinad(session_multihost, request):
+    """ class fixture to join AD using realm """
+    client = sssdTools(session_multihost.client[0], session_multihost.ad[0])
+    client.disjoin_ad()  # Make sure system is disjoined from AD
+    kinit = "kinit Administrator"
+    ad_password = session_multihost.ad[0].ssh_password
+    realm_output = client.join_ad()
+    try:
+        session_multihost.client[0].service_sssd('restart')
+    except SSSDException:
+        cmd = 'cat /etc/sssd/sssd.conf'
+        session_multihost.client[0].run_command(cmd)
+        journal = 'journalctl -x -n 150 --no-pager'
+        session_multihost.client[0].run_command(journal)
```

Comment:
Again, is this relying on the ci/running framework to output results from debug mode?  If not, it may need some slight modification to print/log results as expected.